podvm/README.md

```diff
@@ -0,0 +1,128 @@
+the instructions on the next sections if you need to build your own image with changes to meet your requirements. Otherwise you can
+find [here](../docs/consuming-prebuilt-podvm-images.md) information on how to consume pre-built images.
+
+# How to build on locally
+
+In order to build locally you must [packer](https://www.packer.io/) and QEMU installed in your host. Apart from those, it requires the source trees and software mentioned on the [developer's guide](docs/DEVELOPMENT.md).
+
+Run the following commands to build locally:
+
+```
+$ export CLOUD_PROVIDER=[aws|azure|ibmcloud|libvirt]
+$ make binaries
+$ make image
+```
+
+# How to build within container
+
+This directory contains dockerfiles to build the podvm image entirely within container so that it only requires docker or podman installed on the host.
+
+In general it is needed to follow these steps:
+
+1. Build a builder container image for a given Linux distribution
+1. Build the podvm container image for a given cloud provider
+1. Extract the podvm image (qcow2 file) from the container image
+
+The next sections describe that process in details. Note that although the following examples use docker, it can be carried out with podman too.
+
+## Building a builder image
+
+The builder image packages the cloud-api-adaptor and Kata Containers sources as well as the softwares to build
+the binaries (e.g. *kata-agent* and *agent-protocol-forwarder*) that should be installed in the podvm image.
+
+The builder image is agnostic to cloud providers in the sense that one can be used to buil for multiple providers, however it is
+dependent of the Linux distribution the image is built for. Therefore, in this directory you will find dockerfiles to each supported distributions, which are currently Ubuntu 20.04 ([Dockerfile.podvm_builder](./Dockerfile.podvm_builder)), CentOS Stream 8 ([Dockerfile.podvm_builder.centos](./Dockerfile.podvm_builder.centos)), and RHEL 8.7 ([Dockerfile.podvm_builder.rhel](./Dockerfile.podvm_builder.rhel)).
+
+As an example, to build the builder image for Ubuntu, run:
+
+```
+$ docker build -t podvm_builder -f Dockerfile.podvm_builder .
+```
+
+Use `--build-arg` to pass build arguments to docker to overwrite default values if needed. Following are the arguments
+currently accepted:
+
+|Argument|Default value|Description|
+|--------|-------------|-----------|
+|CAA\_SRC |https://github.com/confidential-containers/cloud-api-adaptor | The cloud-api-adaptor source repository |
+|CAA\_SRC\_BRANCH|staging| cloud-api-adaptor repository branch |
+|KATA\_SRC | https://github.com/kata-containers/kata-containers | The Kata Containers source repository |
+|KATA\_SRC\_BRANCH | CCv0 | The Kata Containers repository branch |
+|GO\_VERSION | 1.18.7 | Go version |
+|PROTOC\_VERSION | 3.11.4 | [Protobuf](https://github.com/protocolbuffers/protobuf) version |
+|RUST\_VERSION | 1.66.0 | Rust version |
```

Comment:
I think we will be able to drop this in favor of allowing to pass CAA_SRC/BRANCH and KATA_SRC/BRANCH to the build containers (and there override the repos) BUT we currently don't have this option so i guess it's fine :)